rstudio/rstudio
# Fancy splash screen

### Intent

Addresses #6876

Note: I only changed the splash screen. The issue also mentioned potentially adding a Posit icon to the About dialog, and showing something on the main window background before the UI loads; if we want either of those I suggest we open separate issue(s).

### Approach

- using new svg in place of the old round "R" one
- use cmake to inject build version and year; on a dev build this will just have a hard coded "Development Build" in place of a version number
    - after doing a package build of Desktop you'll end up with a modified "splash.html" which can be reverted; could do some extra script work to clean this up but it's a fairly rare event and not a problem on the official builds
    - this PR includes two almost identical copies of splash.html; the one used as input to cmake (with version and year placeholders), and the one used in dev builds when cmake hasn't been run in the desktop project
- added "Show Splash Screen" command (command-palette only, and desktop only) to show splash screen while RStudio is running, mainly for testing and easier screen shots; hit any key to close the splash screen

<img width="731" alt="versioned-build" src="https://github.com/user-attachments/assets/fdb5a1a0-dc22-429a-8c50-f84dd6cb9893">

<img width="716" alt="dev-build" src="https://github.com/user-attachments/assets/e5513e1c-0b1f-40d4-a02d-cce553dc7c6d">

### Automated Tests

NA

### QA Notes

I shortened the delay before showing the splash screen so it will appear more often, and also added a bit of a delay when hiding it (trying to avoid having it appear and close too quicky, i.e. flashing without actually being able to read it).

Main things to test are that the version numbering is correct, and also if we should adjust the delays, currently:

- delay before showing: 150 ms (used to be 500 ms)
- minimum display time before hiding: 500 ms

In addition to testing at startup, also try starting new sessions (Session / 

### Documentation

NA

### Checklist
